frontend/src/core/islands/main.ts

```diff
@@ -0,0 +1,139 @@
+/* Copyright 2024 Marimo. All rights reserved. */
+import "./islands.css";
+import "iconify-icon";
+
+import { initializePlugins } from "@/plugins/plugins";
+import { createMarimoFile, parseMarimoIslandApps } from "./parse";
+import { IslandsPyodideBridge } from "./bridge";
+import { store } from "../state/jotai";
+import {
+  createNotebookActions,
+  notebookAtom,
+  notebookReducer,
+} from "../cells/cells";
+import { toast } from "@/components/ui/use-toast";
+import { renderHTML } from "@/plugins/core/RenderHTML";
+import { logNever } from "@/utils/assertNever";
+import { jsonParseWithSpecialChar } from "@/utils/json/json-parser";
+import { FUNCTIONS_REGISTRY } from "../functions/FunctionRegistry";
+import {
+  handleKernelReady,
+  handleCompletedRun,
+  handleInterrupted,
+  handleRemoveUIElements,
+  handleCellOperation,
+} from "../kernel/handlers";
+import { queryParamHandlers } from "../kernel/queryParamHandlers";
+import { Logger } from "@/utils/Logger";
+import { Functions } from "@/utils/functions";
+import { defineCustomElement } from "../dom/defineCustomElement";
+import { MarimoIslandElement } from "./components/web-components";
+import { RuntimeState } from "../kernel/RuntimeState";
+
+/**
+ * Main entry point for the js bundle for embedded marimo apps.
+ */
+
+/**
+ * Initialize the Marimo app.
+ */
+export async function initialize() {
+  // Add 'marimo' class name to all `marimo-island` elements.
+  const islands = document.querySelectorAll<HTMLElement>(
+    MarimoIslandElement.tagName,
+  );
+  for (const island of islands) {
+    island.classList.add(MarimoIslandElement.styleNamespace);
+  }
+
+  // This will display all the static HTML content.
+  initializePlugins();
+
+  // Find all "applications" which consist of snippets of code that are inside <marimo-island> tags.
+  const apps = parseMarimoIslandApps();
+
+  // Initialize each app
+  for (const app of apps) {
```

Comment:
pyodide is pretty heavy, but allowing for multiple apps makes me wonder if marimo could be used to ship widgets like carousels etc..

Probably too slow for anything useful- but interesting thought

Reply:
this is a bit of a relic, but i dont want to remove it yet. we may support multi-app, but this may be a backend thing and the FE may not need to know about it. the use case would be if you had docs or a tutorial and you didnt want to rename each `df` or intentionally wanted to have duplicate `imports` on the same page. its a bit unclear, but there is likely some edge-case use cases, so want to keep an eye open for it.

Reply:
its not tested either (likely wont work since cell's are not unique), but maybe something ill add or remove before moving out of experimental